### PR TITLE
[WC-1716] Issue dg inf loop

### PR DESCRIPTION
**Pull request by iobuhov**
### Pull request type

Bug fix (non-breaking change which fixes an issue)

Breaking change (fix or feature that would cause existing functionality to change)

---

### Description
This PR change how filters react on changes to the defaultValue attribute. Now we use value from this attribute only as initial value and any subsequent change is ignored.

### What should be covered while testing?

Filter widgets - rendering and behavior.

You need a datagrid wrapped with dataview, which is used for storing and managing default values for the filters.

Current behavior:
changing filter value or showing/hiding filter through column selector cause multiple rerenders, poor performance and rendering erros in filter widgets.

Expected:
Changing filter value or hiding/showing filter should not cause any observable flickering or performance drops.

### CODE REVIEW

**Review by samuelreichert — DISMISSED** [review]
LGTM

**Review by rahmanunver — APPROVED** [review]

File: packages/pluggableWidgets/datagrid-text-filter-web/src/features/filter-state.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    useEffect(() => {
+        const { current: prevState } = prevStateRef;
+        if (state.type !== prevState.type) {
+            stableDispatch(state.inputValue, state.type);
+            inputRef.current?.focus();
+        } else if (state.inputValue !== prevState.inputValue) {
+            stableDispatchDelayed(state.inputValue, state.type);
+        }
+        prevStateRef.current = state;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-text-filter-web/src/features/filter-state.ts:51]: Nice question!
Well, this is my intent. 
1. The idea of this `useEffect` is to react _only_ to state changes.
2. Both, `stableDispatch` and `stableDispatchDelayed` are "stable" -- they both use same reference on each render and strictly equal. That means it not worth to add them to deps as they never change during component lifecycle.

**Review by leonardomendix — APPROVED** [review]

**Review by leonardomendix — DISMISSED** [review]

File: packages/pluggableWidgets/datagrid-text-filter-web/src/DatagridTextFilter.tsx [hunk 2]
```diff
@@ -77,24 +77,26 @@ export default function DatagridTextFilter(props: DatagridTextFilterContainerPro
                     return <Alert bootstrapStyle="danger">{errorMessage}</Alert>;
                 }
 
+                if (props.defaultValue && props.defaultValue.status === "loading") {
```
Comment (samuelreichert) [inline packages/pluggableWidgets/datagrid-text-filter-web/src/DatagridTextFilter.tsx:80]: I have one suggestion, maybe you could make it shorter by:
```suggestion
                if (props.defaultValue?.status === "loading") {
```

**Review by samuelreichert — DISMISSED** [review]
LGTM

File: packages/pluggableWidgets/datagrid-text-filter-web/src/DatagridTextFilter.tsx (hunk 2 above)
Comment (iobuhov) [reply packages/pluggableWidgets/datagrid-text-filter-web/src/DatagridTextFilter.tsx]: Done

**Review by leonardomendix — APPROVED** [review]

**Review by samuelreichert — APPROVED** [review]

**Pull request merged**